docs/features/search/declarative-integration.md

```diff
@@ -0,0 +1,229 @@
+
+When a Backstage adopter doesn't want to use the custom `TechDocs` search result item after installing the `TechDocs` plugin, they could disable it via Backstage configuration file:
+
+```yaml
+# app-config.yaml
+app:
+  experimental:
+    packages: 'all'
+  extensions:
+    - plugin.search.result.item.techdocs: false # ✨
+```
+
+Because a configuration schema was provided to the extension factory, Backstage adopters will be able to customize `TechDocs` search results **line clamp** that defaults to 3 and also **disable automatic analytic events tracking**:
+
+```yaml
+# app-config.yaml
+app:
+  experimental:
+    packages: 'all' # enable packages auto discovery
+  extensions:
+    - plugin.search.result.item.techdocs:
+        config: # ✨
+          noTrack: true
+          lineClamp: 3
+```
+
+The `createSearchResultItemExtension` function return an Backstage's extension representation as follows:
+
+```ts
+{
+  "$$type": "@backstage/Extension", // [1]
+  "id": "plugin.search.result.item.techdocs", // [2]
+  "at": "plugin.search.page/items", // [3]
+  "inputs": {} // [4️]
+  "output": { // [5️]
+    "item": {
+      "$$type": "@backstage/ExtensionDataRef",
+      "id": "plugin.search.result.item.data",
+      "config": {}
+    }
+  },
+  "configSchema": { // [6️]
+    "schema": {
+      "type": "object",
+      "properties": {
+        "noTrack": {
+          "type": "boolean",
+          "default": false
+        },
+        "lineClamp": {
+          "type": "number",
+          "default": 5
+        }
+      },
+      "additionalProperties": false,
+      "$schema": "http://json-schema.org/draft-07/schema#"
+    }
+  },
+  "disabled": false, // [7️]
+}
+```
+
+In this object, you can see exactly what will happen once the custom extension is installed:
+
+- **[1] $$type**: declares that the object represents an extension;
+- **[2] id**: Is an unique identification for the extension, the `plugin.search.result.item.techdocs` is the key used to configure the extension in the `app-config.yaml` file;
+- **[3] at**: It represents the extension attachment point, so the value `plugin.search.page/items` says that the `TechDocs`'s search result item output will be injected as input on the "items" attachment expected by the search page extension;
+- configSchema: represents the `TechDocs` search result item configuration definition, this is the same schema that adopters will use for customizing the extension via `app-config.yaml` file;
+- **[4] inputs**: in this case is an empty object because this extension doesn't expect inputs;
+- **[5] output**: Is an abject that represents the artifact produced by the TechDocs result item extension, on the example, it is a react component reference;
```

Comment:
do you mean "is an object"?

Reply:
Fixed 🙏🏻 : https://github.com/backstage/backstage/pull/20328/commits/8558c919cb0ee555b420cbc09d2a1de33be7917d